compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -488,9 +488,15 @@ public void visit(BLangVariable varNode) {
         // Analyze the init expression
         BLangExpression rhsExpr = varNode.expr;
         if (rhsExpr == null) {
-            if (lhsType.tag == TypeTags.ARRAY && ((BArrayType) lhsType).state == BArrayState.OPEN_SEALED) {
-                dlog.error(varNode.pos, DiagnosticCode.INVALID_USAGE_OF_SEALED_TYPE,
-                        "right hand side array literal expected");
+            if (lhsType.tag == TypeTags.ARRAY) {
+                BType type = lhsType;
+                do {
+                    if (((BArrayType) type).state == BArrayState.OPEN_SEALED) {
+                        dlog.error(varNode.pos, DiagnosticCode.INVALID_USAGE_OF_SEALED_TYPE, "array not initialized");
```

Comment:
Lets make the error message includes `array not initialized` part as well.

And lets create separate diagnostic codes, for the separate errors, even though some part of the error message is identical.